Can't compile on Vala 0.48.6

Hi! following the documented installation steps I'm getting the following output:

```
[1/28] Compiling Vala source ../src/collections/LinkedHashMap.vala ....c/utils/TextColorUtil.vala ../src/Application.vala ../src/Main.vala.
FAILED: vanat@exe/src/collections/LinkedHashMap.c vanat@exe/src/commands/AboutCommand.c vanat@exe/src/commands/CommandFacade.c vanat@exe/src/commands/CreateProjectCommand.c vanat@exe/src/commands/HelpCommand.c vanat@exe/src/commands/InitCommand.c vanat@exe/src/commands/InstallCommand.c vanat@exe/src/commands/RemoveCommand.c vanat@exe/src/commands/RequireCommand.c vanat@exe/src/commands/UpdateCommand.c vanat@exe/src/commands/VanatCommand.c vanat@exe/src/commands/VersionCommand.c vanat@exe/src/configs/Constants.c vanat@exe/src/exceptions/FileOrDirectoryNotFoundException.c vanat@exe/src/exceptions/IllegalArgumentException.c vanat@exe/src/exceptions/IOException.c vanat@exe/src/exceptions/JsonException.c vanat@exe/src/exceptions/NullReferenceException.c vanat@exe/src/vjson/VanatJson.c vanat@exe/src/vjson/VPackagistJson.c vanat@exe/src/utils/ConsoleUtil.c vanat@exe/src/utils/FileUtil.c vanat@exe/src/utils/StringUtil.c vanat@exe/src/utils/TextColorUtil.c vanat@exe/src/Application.c vanat@exe/src/Main.c 
valac -C --debug --debug --vapidir /home/devnix/tmp/vanat/vapi --pkg json-glib-1.0 --pkg libsoup-2.4 --pkg libarchive --pkg gio-2.0 --pkg gee-0.8 --pkg glib-2.0 --pkg gobject-2.0 --color=always --directory vanat@exe --basedir ../ --vapidir=/home/devnix/tmp/vanat/vapi/ ../src/collections/LinkedHashMap.vala ../src/commands/AboutCommand.vala ../src/commands/CommandFacade.vala ../src/commands/CreateProjectCommand.vala ../src/commands/HelpCommand.vala ../src/commands/InitCommand.vala ../src/commands/InstallCommand.vala ../src/commands/RemoveCommand.vala ../src/commands/RequireCommand.vala ../src/commands/UpdateCommand.vala ../src/commands/VanatCommand.vala ../src/commands/VersionCommand.vala ../src/configs/Constants.vala ../src/exceptions/FileOrDirectoryNotFoundException.vala ../src/exceptions/IllegalArgumentException.vala ../src/exceptions/IOException.vala ../src/exceptions/JsonException.vala ../src/exceptions/NullReferenceException.vala ../src/vjson/VanatJson.vala ../src/vjson/VPackagistJson.vala ../src/utils/ConsoleUtil.vala ../src/utils/FileUtil.vala ../src/utils/StringUtil.vala ../src/utils/TextColorUtil.vala ../src/Application.vala ../src/Main.vala
json-glib-1.0.vapi:241.6-241.15: warning: [Deprecated] is deprecated. Use [Version (deprecated = true, deprecated_since = "", replacement = "")]
json-glib-1.0.vapi:252.6-252.15: warning: [Deprecated] is deprecated. Use [Version (deprecated = true, deprecated_since = "", replacement = "")]
../src/collections/LinkedHashMap.vala:63.17-63.27: error: Return: Cannot convert from `Gee.Set<K>?' to `Gee.Set<weak K>'
                return keys;
                ^^^^^^^^^^^
../src/collections/LinkedHashMap.vala:79.17-79.29: error: Return: Cannot convert from `Gee.Collection<K>?' to `Gee.Collection<weak V>'
                return values;
                ^^^^^^^^^^^^^
../src/utils/FileUtil.vala:113.28-113.92: error: 1 extra arguments for `Archive.Result Archive.Read.read_data_block (out weak uint8[], out Archive.int64_t)'
                    while (reader.read_data_block(out buffer, out buffer_length, out offset) != Archive.Result.EOF) {
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../src/utils/FileUtil.vala:114.47-114.52: error: Argument 1: Cannot convert from `void*' to `unowned uint8[]'
                        disk.write_data_block(buffer, buffer_length, offset);
                                              ^^^^^^
../src/utils/FileUtil.vala:114.25-114.76: error: 1 extra arguments for `ssize_t Archive.Write.write_data_block (uint8[], Archive.int64_t)'
                        disk.write_data_block(buffer, buffer_length, offset);
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 5 error(s), 2 warning(s)
ninja: build stopped: subcommand failed.
```

--- a/src/collections/LinkedHashMap.vala
+++ b/src/collections/LinkedHashMap.vala
@@ -33,12 +33,12 @@ namespace Vanat.Collections {
         /**
          * 
          */
-        private weak Gee.Set<K> _keyset;
+        private weak Gee.Set<weak K> _keyset;
 
         /**
          * 
          */
-        private weak Gee.Collection<V> _values;
+        private weak Gee.Collection<weak V> _values;
 
         /**
          * 
@@ -49,14 +49,14 @@ namespace Vanat.Collections {
          * [_keys description]
          * @type {ArrayList}
          */
-        internal Gee.ArrayList<K> _keys = new Gee.ArrayList<K>();
+        internal Gee.ArrayList<weak K> _keys = new Gee.ArrayList<weak K>();
 
 
         public new Gee.Set<unowned K> keys {
             owned get {
-                Gee.Set<K> keys = _keyset;
+                Gee.Set<weak K> keys = _keyset;
                 if (_keyset == null) {
-                    keys = new KeySet<K> (_keys);
+                    keys = new KeySet<weak K> (_keys);
                     _keyset = keys;
                     keys.add_weak_pointer ((void**) (&_keyset));
                 }
@@ -69,7 +69,7 @@ namespace Vanat.Collections {
          */
         public new Gee.Collection<unowned V> values {
             owned get {
-                Gee.Collection<K> values = _values;
+                Gee.Collection<weak K> values = _values;
                 if (_values == null) {
                     
                     values = new ValueCollection<K,V> (this);
@@ -295,7 +295,7 @@ namespace Vanat.Collections {
             /**
              * 
              */
-            protected Gee.Iterator<K> _keys;
+            protected Gee.Iterator<weak K> _keys;
 
             /**
              * [ValueIterator description]
@@ -454,7 +454,7 @@ namespace Vanat.Collections {
             /**
              * 
              */
-            protected Gee.Iterator<K> _keys;
+            protected Gee.Iterator<weak K> _keys;
 
             /**
              * 

--- a/src/utils/FileUtil.vala
+++ b/src/utils/FileUtil.vala
@@ -110,8 +110,8 @@ namespace Vanat.Utils {
                 Posix.off_t offset;
 
                 if (entry.size () > 0) {
-                    while (reader.read_data_block(out buffer, out buffer_length, out offset) != Archive.Result.EOF) {
-                        disk.write_data_block(buffer, buffer_length, offset);
+                    while (reader.read_data_block(out buffer, out offset) != Archive.Result.EOF) {
+                        disk.write_data_block((uint8[]) buffer, offset);
                     }
                 }
             }

